lib/get-esm-exports.js

```diff
@@ -14,81 +17,187 @@ function warn (txt) {
-        } else {
-          exportedNames.add('*')
+        path.replace(variableDeclaration)
+  
+        const newExportDefaultDeclaration = {
+          type: 'ExportDefaultDeclaration',
+          declaration: {
+            type: 'Identifier',
+            name: iitmRenamedExport,
+          },
         }
-        break
-      default:
-        warn('unrecognized export type: ' + node.type)
+        path.insertAfter(newExportDefaultDeclaration)
+      } else {
+        node.declaration.id = { type: 'Identifier', name: iitmRenamedExport }
+      }
+  
+      exportAlias['default'] = iitmRenamedExport
+      this.traverse(path)
+    },
+  
+    visitExportAllDeclaration(path) {
+      const node = path.node
+      const exportedName = node.exported ? node.exported.name : '*'
+      exportAlias[exportedName] = exportedName
+      this.traverse(path)
+    },
+  
+    visitExportSpecifier(path) {
+      const node = path.node
+      exportSpecifierNames.add(node.local.name)
+  
+      if (node.exported.name) {
+        exportAlias[node.exported.name] = node.local.name
+      } else if (node.exported.value) {
+        exportAlias[node.exported.value] = node.local.name
+      } else {
+        warn('unrecognized specifier export: ' + node.exported)
+      }
+
+      this.traverse(path)
+    },
+  })
+  
+  if (exportSpecifierNames.size !== 0 && generate) {
+    convertExportSpecifierToLet(exportSpecifierNames, ast)
+  }
+
+  if (generate) {
+    return {
+      exportAlias: exportAlias,
+      code: recast.print(ast).code
     }
   }
-  return Array.from(exportedNames)
+
+  return Object.keys(exportAlias)
+}
+
+function convertExportSpecifierToLet(exportSpecifierNames, ast) {
+  recast.visit(ast, {
+    visitVariableDeclaration(path) {
+      const declaration = path.node
+      if (declaration.kind === 'const') {
+        for (const declarator of declaration.declarations) {
+          const variableName = declarator.id.name
+          if (exportSpecifierNames.has(variableName)) {
+            declaration.kind = 'let'
+          }
+        }
+      }
+      this.traverse(path)
+    },
+  })
 }
 
-function parseDeclaration (node, exportedNames) {
-  switch (node.declaration.type) {
+function parseDeclaration(declaration, exportAlias) {
+  switch (declaration.type) {
     case 'FunctionDeclaration':
-      exportedNames.add(node.declaration.id.name)
+      exportAlias[declaration.id.name] = declaration.id.name
       break
     case 'VariableDeclaration':
-      for (const varDecl of node.declaration.declarations) {
-        parseVariableDeclaration(varDecl, exportedNames)
+      for (const varDecl of declaration.declarations) {
+        if (declaration.kind === 'const') {
+          declaration.kind = 'let'
+        }
+        parseVariableDeclaration(varDecl, exportAlias)
       }
       break
     case 'ClassDeclaration':
-      exportedNames.add(node.declaration.id.name)
+      exportAlias[declaration.id.name] = declaration.id.name
```

Comment:
This is now the same case as `FunctionDeclaration`, so they can be handled together.